### DISCUSSION TITLE
No WiFi after SD issue

### DISCUSSION THREAD

**Discussion by tniccum21** [thehighnibble/thehighnibble.github.io#183] [IMSAI8080esp replica]
Hey all, I'm working my way through getting my machine running again... It ran on my shelf without issue for a couple years...

A few days ago I looked up at my shelf and all my retros were purring away with blinky lights, but the Imsai was dark... 

So tonight I pulled it down and as I suspected the SD card had died.

So... I downloaded the software, created a new SD card... Connecting a serial monitor, i see the stream below - not seeing any obvious errors.

The lights are on, when I turn it to PWR ON, but no IMSAI shows up on my wifi...

One question:  is there a limit on SD size?  I had a 64GB laying around and used it... but I know on some of these there are limits.

Thanks,

Tom

--Here's the boot up log--

rst:0x1 (POWERON_RESET),boot:0x3f (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0018,len:4
load:0x3fff001c,len:4492
load:0x40078000,len:13324
ho 0 tail 12 room 4
load:0x40080400,len:3796
entry 0x40080624
[0;32mI (359) psram: This chip is ESP32-PICO[0m
[0;32mI (359) spiram: Found 64MBit SPI RAM device[0m
[0;32mI (359) spiram: SPI RAM mode: flash 80m sram 80m[0m
[0;32mI (362) spiram: PSRAM initialized, cache is in low/high (2-core) mode.[0m
[0;32mI (369) cpu_start: Pro cpu up.[0m
[0;32mI (373) cpu_start: Application information:[0m
[0;32mI (378) cpu_start: Project name:     imsaisim_esp32[0m
[0;32mI (384) cpu_start: App version:      v1.11.0[0m
[0;32mI (388) cpu_start: Compile time:     Mar 19 2023 00:37:52[0m
[0;32mI (395) cpu_start: ELF file SHA256:  ea07ca307884de9c...[0m
[0;32mI (401) cpu_start: ESP-IDF:          v4.0.2-5-ged96bbd56c[0m
[0;32mI (407) cpu_start: Starting app cpu, entry point is 0x40081d10[0m
[0;32mI (0) cpu_start: App cpu up.[0m
[0;32mI (903) spiram: SPI SRAM memory test OK[0m
[0;32mI (938) heap_init: Initializing. RAM available for dynamic allocation:[0m
[0;32mI (938) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM[0m
[0;32mI (940) heap_init: At 3FFC7260 len 00018DA0 (99 KiB): DRAM[0m
[0;32mI (946) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM[0m
[0;32mI (953) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM[0m
[0;32mI (959) heap_init: At 4009EF78 len 00001088 (4 KiB): IRAM[0m
[0;32mI (965) cpu_start: Pro cpu start user code[0m
[0;32mI (970) spiram: Adding pool of 3621K of external SPI memory to heap allocator[0m
[0;32mI (991) spi_flash: detected chip: gd[0m
[0;32mI (991) spi_flash: flash io: qio[0m
[0;32mI (991) cpu_start: Starting scheduler on PRO CPU.[0m
[0;32mI (0) cpu_start: Starting scheduler on APP CPU.[0m
[0;32mI (1059) nvs: Reading settings from NVS [0m
[0;32mI (1060) nvs: settings = 00000958[0m
[0;32mI (1060) esp32_boot: Log Level set to NONE[0m
[0;32mI (1689) phy: phy_version: 4390, 6b3c1f2, Sep 10 2020, 15:09:07, 0, 0[0m

#######  #####    ###            #####    ###   #     #
     #  #     #  #   #          #     #    #    ##   ##
    #   #     # #     #         #          #    # # # #
   #     #####  #     #  #####   #####     #    #  #  #
  #     #     # #     #               #    #    #     #
 #      #     #  #   #          #     #    #    #     #
#######  #####    ###            #####    ###   #     #

Release 1.38-dev, Copyright (C) 1987-2022 by Udo Munk
ESP32 IMSAI 8080 Simulation Release 1.19 (ESP32)
Copyright (C) 2008-2022 by Udo Munk & David McNaughton

CPU speed is 4 MHz, CPU executes undocumented instructions

RAM 0000H - FFFFH

MPU-B Banked ROM/RAM enabled

MMU has 7 additional RAM banks of 48 KB

SIO PORT MAP:

SIO1.portA = WEBTTY UART0  

SIO1.portB = VIOKBD  

SIO2.portA = WEBPTR UART1  

SIO2.portB = MODEM  

LPT = WEBLPT FILE

**Comment by tniccum21** [comment]
Additional note:

I tried to access/check/set the NVRAM configuration, but that's not working...

I power off, hold up examine, hit reset, release examine.  Nothing.  No lights whatsoever.

HOWEVER... I now see imsai (but not imsai8080) on my network....

If I reset from ere, I get the run light lit, and most of the lights are on and something appears to be running.

**Comment by tniccum21** [comment]
After rebooting (pressing "EN" on ESP) i'm now getting wifi messages (imsai showing up on newtorks, but not imsai8080 still):

Release 1.38-dev, Copyright (C) 1987-2022 by Udo Munk
ESP32 IMSAI 8080 Simulation Release 1.19 (ESP32)
Copyright (C) 2008-2022 by Udo Munk & David McNaughton

CPU speed is 4 MHz, CPU executes undocumented instructions

RAM 0000H - FFFFH
MPU-B Banked ROM/RAM enabled
MMU has 7 additional RAM banks of 48 KB

SIO PORT MAP:
SIO1.portA = WEBTTY UART0  
SIO1.portB = VIOKBD  
SIO2.portA = WEBPTR UART1  
SIO2.portB = MODEM  
LPT = WEBLPT FILE

[H[2JE (70787) wifi:esf_buf: t=2 l=216 max:32, alloc:32 no eb, TXQ_BLOCK=4000
E (70790) wifi:esf_buf: t=2 l=216 max:32, alloc:32 no eb, TXQ_BLOCK=4000
E (70793) wifi:esf_buf: t=2 l=216 max:32, alloc:32 no eb, TXQ_BLOCK=4000
E (70797) wifi:esf_buf: t=2 l=216 max:32, alloc:32 no eb, TXQ_BLOCK=4000
E (70803) wifi:esf_buf: t=2 l=216 max:32, alloc:32 no eb, TXQ_BLOCK=2000
E (70809) wifi:esf_buf: t=2 l=216 max:32, alloc:32 no eb, TXQ_BLOCK=4000

**Reply by tniccum21** [reply]
OK I also updated the firmware to 1.13.0 in case the firmware was out of compatibility with the newest image, etc.
Boots OK, and now i figured out the right reset button to push, 
 the NVRAM setup works. 

However after PWR ON, RUN, i see the imsai wifi, but not the imsai8080.

It seems that on bootup there is no imsai on wifi, but if II enter the NVRAM setup, and exit - then imsai shows up....  

I upped the logging to "info" level and now detecting some issues - also in a boot loop, where it's trying to reboot after that Guru Meditation Error core dump.

===================

Boot log is showing 1.13.0 as the firmware version.

ELF file SHA256: a195df7f0fdfe351

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x3f (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:4940
ho 0 tail 12 room 4
load:0x40078000,len:15120
load:0x40080400,len:3764
entry 0x40080620
[0;32mI (307) psram: This chip is ESP32-PICO[0m
[0;32mI (307) spiram: Found 64MBit SPI RAM device[0m
[0;32mI (307) spiram: SPI RAM mode: flash 80m sram 80m[0m
[0;32mI (310) spiram: PSRAM initialized, cache is in low/high (2-core) mode.[0m
[0;32mI (317) cpu_start: Pro cpu up.[0m
[0;32mI (321) cpu_start: Starting app cpu, entry point is 0x40081c60[0m
[0;32mI (313) cpu_start: App cpu up.[0m
[0;32mI (812) spiram: SPI SRAM memory test OK[0m
[0;32mI (854) cpu_start: Pro cpu start user code[0m
[0;32mI (854) cpu_start: cpu freq: 240000000[0m
[0;32mI (854) cpu_start: Application information:[0m
[0;32mI (857) cpu_start: Project name:     imsaisim_esp32[0m
[0;32mI (863) cpu_start: App version:      V1.13.0[0m
[0;32mI (868) cpu_start: Compile time:     Sep  1 2024 06:11:03[0m
[0;32mI (874) cpu_start: ELF file SHA256:  a195df7f0fdfe351...[0m
[0;32mI (880) cpu_start: ESP-IDF:          v4.4.4-1-gadda465dc3-dirty[0m
[0;32mI (886) heap_init: Initializing. RAM available for dynamic allocation:[0m
[0;32mI (894) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM[0m
[0;32mI (900) heap_init: At 3FFBC570 len 00023A90 (142 KiB): DRAM[0m
[0;32mI (906) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM[0m
[0;32mI (912) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM[0m
[0;32mI (919) heap_init: At 4009DC98 len 00002368 (8 KiB): IRAM[0m
[0;32mI (925) spiram: Adding pool of 3629K of external SPI memory to heap allocator[0m
[0;32mI (933) spi_flash: detected chip: gd[0m
[0;32mI (937) spi_flash: flash io: qio[0m
[0;32mI (942) cpu_start: Starting scheduler on PRO CPU.[0m
[0;32mI (0) cpu_start: Starting scheduler on APP CPU.[0m
[0;32mI (1019) nvs: Reading settings from NVS [0m
[0;32mI (1020) nvs: settings = 00000006[0m
[0;32mI (1020) esp32_boot: Log Level set to INFO[0m
[0;32mI (1022) esp32_hardware: Shell Running on Core#0[0m
[0;32mI (1508) esp32_hardware: CP-A Found - Power Switch 1[0m
[0;32mI (1508) esp32_hardware: DIP/MEM Switches: 00[0m
[0;32mI (1508) esp32_hardware: Initializing uSD card[0m
[0;32mI (1513) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 [0m
[0;32mI (1566) esp32_hardware: uSD card successfully detected and mounted with the following details:[0m
Name: SD   
Type: SDHC/SDXC
Speed: 40 MHz
Size: 59638MB
[0;32mI (1570) esp32_update: Checking for image update package[0m
[0;32mI (1577) esp32_boot: OPENING BOOT.CONF[0m
[0;32mI (1581) esp32_hardware: Initialise UART1 to default configuration[0m
I (1590) wifi:wifi driver task: 3ffc4684, prio:23, stack:3584, core=0
[0;32mI (1594) system_api: Base MAC address is not set[0m
[0;32mI (1599) system_api: read default base MAC address from EFUSE[0m
I (1619) wifi:wifi firmware version: 6567a16
I (1619) wifi:wifi certification version: v7.0
I (1619) wifi:config NVS flash: enabled
I (1620) wifi:config nano formating: disabled
I (1624) wifi:Init data frame dynamic rx buffer num: 32
I (1629) wifi:Init management frame dynamic rx buffer num: 32
I (1634) wifi:Init management short buffer num: 32
I (1639) wifi:Init dynamic tx buffer num: 32
I (1643) wifi:Init tx cache buffer num: 32
I (1647) wifi:Init static rx buffer size: 1600
I (1651) wifi:Init static rx buffer num: 16
I (1655) wifi:Init dynamic rx buffer num: 32
[0;32mI (1659) wifi_init: rx ba win: 16[0m
[0;32mI (1663) wifi_init: tcpip mbox: 64[0m
[0;32mI (1667) wifi_init: udp mbox: 6[0m
[0;32mI (1671) wifi_init: tcp mbox: 10[0m
[0;32mI (1674) wifi_init: tcp tx win: 11680[0m
[0;32mI (1679) wifi_init: tcp rx win: 11680[0m
[0;32mI (1683) wifi_init: tcp mss: 1460[0m
[0;32mI (1687) wifi_init: WiFi IRAM OP enabled[0m
[0;32mI (1692) wifi_init: WiFi RX IRAM OP enabled[0m
[0;32mI (1697) wifi_init: LWIP IRAM OP enabled[0m
I (1702) wifi:Set ps type: 0

[0;32mI (1704) esp32_hardware: MAC ADDR: d8:a0:1d:5d:be:f5[0m
[0;32mI (1710) esp32_netsrv: AP Mode - Using SSID: imsai (HOSTNAME)[0m
[0;32mI (1719) esp32_netsrv: Starting Wi-Fi now[0m
[0;32mI (1721) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07[0m
I (1815) wifi:mode : softAP (d8:a0:1d:5d:be:f5)
I (1816) wifi:Total power save buffer number: 16
I (1817) wifi:Init max length of beacon: 752/752
I (1817) wifi:Init max length of beacon: 752/752
[0;32mI (1822) esp32_netsrv: EVENT: WIFI_EVENT - 12[0m
[0;32mI (1826) esp32_netsrv: soft-AP started[0m
[0;32mI (1831) dns_server: STARTING DNS SERVER[0m
[0;32mI (1835) start_dns_server: dns server task started[0m
[0;32mI (1841) dns_server: Socket created[0m
[0;32mI (1845) dns_server: Socket bound, port 53[0m
[0;32mI (1850) dns_server: Waiting for data[0m
Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.

Core  0 register dump:
PC      : 0x400014fd  PS      : 0x00060230  A0      : 0x80159aa4  A1      : 0x3ffc2890  
A2      : 0x00000000  A3      : 0xfffffffc  A4      : 0x000000ff  A5      : 0x0000ff00  
A6      : 0x00ff0000  A7      : 0xff000000  A8      : 0x00000000  A9      : 0x3ffc2850  
A10     : 0x00000000  A11     : 0x3ffc2a94  A12     : 0x3ffc2ad4  A13     : 0x3ffaeb4c  
A14     : 0x00000000  A15     : 0x00000001  SAR     : 0x00000004  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000000  LBEG    : 0x400014fd  LEND    : 0x4000150d  LCOUNT  : 0xffffffff  


Backtrace: 0x400014fa:0x3ffc2890 0x40159aa1:0x3ffc28a0 0x4015b111:0x3ffc2bb0 0x40168949:0x3ffc2be0 0x40095109:0x3ffc2c10 0x400dafe4:0x3ffc2c60 0x400da499:0x3ffc2c90 0x400da4ef:0x3ffc2cb0 0x40167c57:0x3ffc2d60 0x40167d24:0x3ffc2db0

**Reply by tniccum21** [reply]
I replaced the ESP32 board, but basically getting exactly the same issue,,, 
Guru Meditation Error: Core 0 panic'ed (LoadProhibited). Exception was unhandled.

Any help troubleshooting this would be more than welcome.

**Comment by dmcnaugh** [comment]
Tom,
it would help greatly if you **email** a fresh copy of the console debug log output from a full 2 cycles of pressing the `EN` reset button on the ESP32 module.

I can then look through that and see if I can identify the problem.

Once we resolve the issue we can post the solution here.

**Comment by tniccum21** [comment]
Thanks David, I just resolved the issue.  

I had multiple things going wrong - the usual case when things just don't work like they should.

1. Something (powersurge? static?) wiped out my SD card... to the point where it was unusable
2. Whatever took out my SD card in the beginning also wiped the Non-volatile RAM settings.   I did not realize this.
3. Part of the difficulty in trouble shooting was that I had my back panel connector connected to the wrong port on the ESP32... moving that helped a lot :)

So after re-setting the PSRAM/non-volatile memory with the correct info, viola... my wifi was back, i could access the web page interface, etc.

SO, I'm back on the air - I just hadn't had time to write up my final findings.

Thanks for checking back in.
